### PR TITLE
Update the JSDocs for the public API

### PR DESCRIPTION
**Pull request by DominicKramer**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -67,15 +67,29 @@ var createLogger = require('./src/logger.js');
...
- * the `report` function property, with hapi via the `hapi` object property or
- * with express via the `express` function property.
- * @function initConfiguration
+ * <p class="notice">
+ *   **This is not an official Google product.** This module is experimental
+ *   and may not be ready for use.  This module uses APIs that may be
+ *   undocumented and are subject to change without notice.
+ * </p>
+ *
+ * This modules provides Stackdriver Error Reporting support for Node.js
```
Comment (kjin) [inline index.js]: modules -> module

Comment (kjin) [inline index.js]: Actually, I think this sentence might be unnecessary altogether.

Comment (DominicKramer) [inline index.js]: Fixed

File: index.js [hunk 2]
```diff
@@ -67,15 +67,29 @@ var createLogger = require('./src/logger.js');
...
+ *   undocumented and are subject to change without notice.
+ * </p>
+ *
+ * This module provides Stackdriver Error Reporting support for Node.js
+ * applications.
+ * [Stackdriver Error Reporting](https://cloud.google.com/error-reporting/) is
+ * a feature of Google Cloud Platform that allows in-depth monitoring and
+ * viewing of errors reported by applications running in almost any environment.
+ *
+ * This module initializes the Error Reporting Middleware and provides an object
```
Comment (matthewloring) [inline index.js]: I don't think Error Reporting Middleware should be capitalized.

File: index.js [hunk 3]
```diff
@@ -67,15 +67,29 @@ var createLogger = require('./src/logger.js');
...
+ * that can be used manually through the `report` function property, with Hapi
+ * via the `hapi` object property or with Express via the `express` function
```
Comment (matthewloring) [inline index.js]: This description of the interface can be shortened if there is a way to connect this constructor with the `Errors` type defined above. Does the `@alias` below do that?

**Comment by DominicKramer** [comment]
This PR has been closed since its contents have been incorporated into the [google-cloud-node/error-reporting](https://github.com/GoogleCloudPlatform/google-cloud-node/tree/master/packages/error-reporting) repository via [PR #2265](https://github.com/GoogleCloudPlatform/google-cloud-node/pull/2265).

**Pull request closed without merging**